lexer/internal/conformance/conformance_test.go

```diff
@@ -92,6 +113,68 @@ func testLexer(t *testing.T, lex lexer.Definition) {
 		// 			{"Whitespace", "\n"},
 		// 			{"End", "EOF"},
 		// 		}},
+		{"CaseInsensitiveSimple", `CITEST:hello aBC world`, []token{
+			{"CaseInsensitiveTest", "CITEST:"},
+			{"Ident", "hello"},
+			{"Whitespace", " "},
+			{"ABCWord", "aBC"},
+			{"Whitespace", " "},
+			{"Ident", "world"},
+		}},
+		{"CaseInsensitiveMixed", `CITEST:hello SeLeCt FROM world where END`, []token{
+			{"CaseInsensitiveTest", "CITEST:"},
+			{"Ident", "hello"},
+			{"Whitespace", " "},
+			{"CIKeyword", "SeLeCt"},
+			{"Whitespace", " "},
+			{"CIKeyword", "FROM"},
+			{"Whitespace", " "},
+			{"Ident", "world"},
+			{"Whitespace", " "},
+			{"CIKeyword", "where"},
+			{"Whitespace", " "},
+			{"Ident", "END"},
+		}},
+		{"OneLiteralAtEnd", `LITTEST:ONE`, []token{
+			{"LiteralTest", "LITTEST:"},
+			{"LITOne", "ONE"},
+		}},
+		{"KeywordLiteralAtEnd", `LITTEST:SELECT`, []token{
+			{"LiteralTest", "LITTEST:"},
+			{"LITKeyword", "SELECT"},
+		}},
+		{"LiteralMixed", `LITTEST:hello ONE test LIKE world`, []token{
+			{"LiteralTest", "LITTEST:"},
```

Comment:
Maybe we could make the test harness strip these automatically? ie. the first token iff it ends with `TEST:`